### PR TITLE
[#47] Add component: sidebar

### PR DESCRIPTION
**Pull request by andreasknoepfle**
Screenshot 2021-12-16 at 16 58 10
Screenshot 2021-12-16 at 16 58 41
Screenshot 2021-12-16 at 16 58 44
Screenshot 2021-12-16 at 16 59 34
Screenshot 2021-12-16 at 16 59 50

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ jobs:
       - run: mix deps.get
       - run: mix format --check-formatted
       - run: mix test
+      - run: mix credo
```
Comment (maltoe) [inline .circleci/config.yml:16]:
```suggestion
      - run: mix credo --strict
```
No?

File: demo/lib/bitstyles_phoenix_demo_web/live/demo_live.ex [hunk 2]
```diff
@@ -1,3 +1,13 @@
 defmodule BitstylesPhoenixDemoWeb.DemoLive do
   use BitstylesPhoenixDemoWeb, :live_view
+
+  def handle_event("put_flash", _, socket) do
```
Comment (maltoe) [inline demo/lib/bitstyles_phoenix_demo_web/live/demo_live.ex:4]:
```suggestion
  def handle_event("show-flash", _, socket) do
```

File: demo/lib/bitstyles_phoenix_demo_web/live/demo_live.html.heex [hunk 3]
```diff
@@ -1,5 +1,6 @@
 <main class="a-content">
   <h1>Live test</h1>
+  <p><%= ui_button "Flash messages", "phx-click": "put_flash" %></p>
```
Comment (maltoe) [inline demo/lib/bitstyles_phoenix_demo_web/live/demo_live.html.heex:3]:
```suggestion
  <p><%= ui_button "Flash messages", "phx-click": "show-flash" %></p>
```

File: lib/bitstyles_phoenix/alpine3/dropdown.ex [hunk 4]
```diff
@@ -3,6 +3,10 @@ defmodule BitstylesPhoenix.Alpine3.Dropdown do
 
   alias BitstylesPhoenix.Component.Dropdown, as: RawDropdown
 
+  @moduledoc """
+  Components for rendering a drowdowns powered by Alpine 3.
```
Comment (maltoe) [inline lib/bitstyles_phoenix/alpine3/dropdown.ex:7]:
```suggestion
  Components for rendering a dropdown menu powered by Alpine 3.
```

File: lib/bitstyles_phoenix/alpine3/sidebar.ex [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+        x-transition:leave-end="is-off-screen"
+        {@small_extra}>
+        <%= assigns[:small_sidebar] && render_block(assigns[:small_sidebar], {@x_data, @small_extra[:id]}) %>
+      </:small_sidebar>
+      <:main {main_extra}>
+        <%= render_block(assigns[:main] || @inner_block, {@x_data, @small_extra[:id]}) %>
+      </:main>
+    </RawSidebar.ui_sidebar_layout>
+    """
+  end
```
Comment (maltoe) [inline lib/bitstyles_phoenix/alpine3/sidebar.ex:201]: i don't understand where `large_sidebar` is gone!?

**Review by maltoe — APPROVED** [review]
nice work 🎉

File: lib/bitstyles_phoenix/alpine3/sidebar.ex (hunk 5 above)
Comment (andreasknoepfle) [reply lib/bitstyles_phoenix/alpine3/sidebar.ex:201]: It is just passed on to the `RawSidebar` via `extra`

**Pull request merged**